Firmware/MotorControl/thermistor.hpp

```diff
@@ -0,0 +1,132 @@
+#ifndef __THERMISTOR_HPP
+#define __THERMISTOR_HPP
+
+#ifndef __ODRIVE_MAIN_H
+#error "This file should not be included directly. Include odrive_main.h instead."
+#endif
+
+class ThermistorCurrentLimiter : public CurrentLimiter {
+public:
+    enum Error_t {
+        ERROR_NONE = 0x0000,
+        ERROR_OVER_TEMP = 0x0001,
+    };
+
+    ThermistorCurrentLimiter() = delete;
+    virtual ~ThermistorCurrentLimiter() = default;
+    ThermistorCurrentLimiter(const ThermistorCurrentLimiter& other) = delete;
+    ThermistorCurrentLimiter(ThermistorCurrentLimiter&& other) noexcept = delete;
+    ThermistorCurrentLimiter& operator=(const ThermistorCurrentLimiter& original) = delete;
+    ThermistorCurrentLimiter& operator=(ThermistorCurrentLimiter&& original) = delete;
```

Comment:
IMO for brevity we can omit these declarations. I mean it's true that it's not allowed to copy/move a `ThermistorCurrentLimiter` object but in other classes (Motor, Axis, ...) we also don't state this explicitly. Btw I think copy/move assignment operators are not autogenerated for classes that contain references (because the reference itself is const).

Reply:
Fixed